var/spack/repos/builtin/packages/alsa-lib/package.py

```diff
@@ -23,6 +24,8 @@ class AlsaLib(AutotoolsPackage):
 
     depends_on('python', type=('link', 'run'), when='+python')
 
+    conflicts('platform=darwin', msg='ALSA only works for Linux')
```

Comment:
Is there any way that we can make this package conflict for all non-linux platforms? It seems to me like that would be the better long-term solution.

Reply:
We can do:
```python
import sys

if not sys.platform.startswith('linux'):
    conflicts()
```
but that may break things when cross-compiling. We can also do:
```python
for platform in ['darwin', 'cray', ...]:
    conflicts('platform=' + platform)
```
but I don't know the full `...` list. I also imagine that ALSA may work on Cray, which is really just Linux plus some compiler wrappers, but I haven't tested this.

Reply:
Since there isn't any easy way to isolate and test the Linux platforms and you have a message about when `alsa` should work, I think your solution is fine for the time being.